src/content/docs/develop/state-management.mdx

```diff
@@ -2,32 +2,180 @@
 title: State Management
 sidebar:
   order: 1
-  badge:
-    text: Stub
-    variant: caution
 ---
 
-import Stub from '@components/Stub.astro';
+In a Tauri application, you often need to keep track of the current state of your application or manage the lifecycle of things associated with it. Tauri provides an easy way to manage the state of your application using the `Manager` API, and read it when commands are called.
 
-<Stub>
+Here is a simple example:
 
-Define a state struct and let Tauri manage it:
+```rust
+use tauri::{Builder, Manager};
+
+struct AppData {
+  welcome_message: &'static str,
+}
+
+// In `fn main()`:
+Builder::default()
+  .setup(|app| {
+    app.manage(AppData {
+      welcome_message: "Welcome to Tauri!",
+    });
+    Ok(())
+  })
+  .run(tauri::generate_context!())
+  .unwrap();
+```
+
+You can later access your state via the `App` instance:
+
+```rust
+let data = app.state::<AppData>();
+```
+
+For more info, including accessing state in commands, see the [Accessing State](#accessing-state) section.
+
+## Mutability
+
+In Rust, values which can be shared between multiple owners (for example, sent to different threads) cannot be mutable. This is to prevent multiple owners from modifying the value at the same time, which can lead to issues such as data races.
+
+To work around this, you can use the standard library's `Mutex` to wrap your state. This allows you to lock the value when you need to modify it, and unlock it when you are done.
```

Comment:
I would make this a bit general like "you can use one of the various techniques to achieve mutability" and link to these Rust book sections https://doc.rust-lang.org/book/ch15-05-interior-mutability.html  and https://doc.rust-lang.org/book/ch16-00-concurrency.html

After that we can make a suggestion like "for this example, we will use `Mutex` which locks the value...etc"